tests/functional/TestSmokeTest/tests/heketi_test.go

```diff
@@ -171,48 +170,33 @@ func HeketiCreateVolumeWithGid(t *testing.T) {
 	volReq.Snapshot.Enable = true
 	volReq.Snapshot.Factor = 1.5
 
-	// Set to the vagrant gid
-	volReq.Gid = 1000
+	volReq.Gid = 2333
 
 	// Create the volume
 	volInfo, err := heketi.VolumeCreate(volReq)
 	tests.Assert(t, err == nil, "expected err == nil, got:", err)
 
-	// SSH into system and create two writers belonging to writegroup gid
-	vagrantexec := na.Use(logger)
+	// SSH into system, create two writers belonging to writegroup gid and
+	// make sure both can write and the sticky group bit is set
+	exec := na.Use(logger)
 	cmd := []string{
-		"sudo groupadd writegroup",
-		"sudo useradd writer1 -G writegroup -p'$6$WBG5yf03$3DvyE41cicXEZDW.HDeJg3S4oEoELqKWoS/n6l28vorNxhIlcBe2SLQFDhqq6.Pq'",
-		"sudo useradd writer2 -G writegroup -p'$6$WBG5yf03$3DvyE41cicXEZDW.HDeJg3S4oEoELqKWoS/n6l28vorNxhIlcBe2SLQFDhqq6.Pq'",
+		"sudo groupadd -f -g 2333 writegroup",
+		"grep -q ^writer1: /etc/passwd || sudo useradd writer1 -G writegroup",
+		"grep -q ^writer2: /etc/passwd || sudo useradd writer2 -G writegroup",
+		"sudo umount /mnt 2>/dev/null || true",
 		fmt.Sprintf("sudo mount -t glusterfs %v /mnt", volInfo.Mount.GlusterFS.MountPoint),
+		"sudo runuser -u writer1 -- touch /mnt/writer1testfile",
+		"sudo runuser -u writer1 -- mkdir /mnt/writer1dir",
+		"sudo runuser -u writer1 -- chmod 770 /mnt/writer1dir",
```

Comment:
The expectation is that it works without chmod. When a pv is created in kubernetes with a gid all the writes from any user who belong to the group must be able to write.

Reply:
according to my tests the sticky bit only sets the group, not the permissions. the chmod was neccessary, or the mode was 2755.

Reply:
I stand corrected. SGID only sets group inheritance not permission inheritance. So we need to check if SGID is sufficient as per kubernetes or if we need something more.

Reply:
This PR addresses the problem the original test had. With that, we can merge the PR.

We will investigate on kubernetes requirements and work on it in a different PR, if required.